### ISSUE TITLE
Build failure (bad linker flags) on Solaris-11

### ISSUE THREAD

**Issue by PHHargrove** [Intrepid/llvm-upc#13]
My Solaris-11/x86-64 builds that were working a while ago are now failing as follows:
```
llvm[2]: Linking Release+Asserts executable llvm-tblgen
ld: fatal: unrecognized option '--'
ld: fatal: use the -z help option for usage information
collect2: error: ld returned 1 exit status
```

The message `ld: fatal: use the -z help option for usage information` is from the Solaris linker, `/usr/bin/ld`, while the `--` argument is probably intended for the Gnu linker.  So, this appears to be related to use of the Gnu-vs-Solaris linker.  Previously I know I needed to ensure /usr/gnu/bin was in my PATH prior to /usr/bin, but that no longer appears sufficient.

There may be a way to get the right behavior via arguments to configure or cmake.
I delayed reporting this for 2 days hoping to find the magical args required, but have not yet found the time to investigate.

**Comment by PHHargrove** [comment]
This now appears to be "pilot error".

Because the system gcc is 4.5, I had built my own gcc-4.9.2 to compile llvm+clang.
That gcc defaulted to using the Solaris linker.
I am now rebuilding gcc-4.9.2 to use the gnu linker instead.
I suspect that will be sufficient to resolve the problem.

I will close once I've confirmed the proposed fix.

**Comment by PHHargrove** [comment]
I now have a successful build of clang-upc (et al.) on Solaris-11/x86-64 with the default (ILP32) ABI.

I needed to *both*
* Ensure gcc was configured with --with-gnu-ld and --with-gnu-as
* Place /usr/gnu/bin ahead of /usr/bin at configure and make time

I still have some issues with trying to build for the LP64 ABI, but which I mean clang and friends as 64-bit executable - I don't expect to use multilib support in clang.  I will update this issue when I have sorted those issues out.
